internal/service/instance/nginx_instance_service.go

```diff
@@ -271,3 +278,36 @@ func isFlag(vals []string) bool {
 func isKeyValueFlag(vals []string) bool {
 	return len(vals) == 2
 }
+
+func getLoadableModules(nginxInfo *Info) (modules []string) {
+	var err error
+	if nginxInfo.ConfigureArgs["modules-path"] != nil {
+		modulePath, ok := nginxInfo.ConfigureArgs["modules-path"].(string)
+		if !ok {
+			slog.Warn("error parsing modules-path")
+			return modules
+		}
+		modules, err = readDirectory(modulePath, ".so")
+		if err != nil {
+			slog.Warn("error reading module dir", "dir", modulePath, "error", err)
+			return modules
+		}
+
+		return modules
+	}
+
+	return modules
+}
+
+func readDirectory(dir, ext string) (files []string, err error) {
+	dirInfo, err := os.ReadDir(dir)
+	if err != nil {
+		return files, fmt.Errorf("unable to read directory %s, %w", dir, err)
```

Comment:
https://github.com/uber-go/guide/blob/master/style.md#error-wrapping
> When adding context to returned errors, keep the context succinct by avoiding phrases like "failed to", which state the obvious and pile up as the error percolates up through the stack:
```suggestion
		return files, fmt.Errorf("read directory %s, %w", dir, err)
```